### PR TITLE
Add set/getSystemAttentionIndicator

### PR DESCRIPTION
**Pull request by lxwinspur**
Added OEM commands for platform and partition to set/get SAI
refer: https://github.com/ibm-openbmc/dev/issues/2120

Signed-off-by: George Liu <liuxiwei@inspur.com>

### CODE REVIEW

**Comment by gtmills** [comment]
Redfish validator pass? 
Where is the OEM schema for this?

**Comment by lxwinspur** [comment]
> Redfish validator pass? 

Yes, Redfish validator passed.

> Where is the OEM schema for this?

https://github.com/ibm-openbmc/bmcweb/blob/1020/static/redfish/v1/schema/OemComputerSystem_v1.xml

**Comment by gtmills** [comment]
> https://github.com/ibm-openbmc/bmcweb/blob/1020/static/redfish/v1/schema/OemComputerSystem_v1.xml

File: redfish-core/lib/systems.hpp [hunk 1]
```diff
@@ -3522,8 +3529,14 @@ inline void requestRoutesSystems(App& app)
                     if (ibmOem)
                     {
                         std::optional<bool> lampTest;
-                        if (!json_util::readJson(*ibmOem, asyncResp->res,
-                                                 "LampTest", lampTest))
+                        std::optional<bool> partitionSAI;
```
Comment (gtmills) [inline redfish-core/lib/systems.hpp:3532]: Can we put these in the ifdef as well?

File: redfish-core/lib/oem/ibm/system_attention_indicator.hpp [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    {
+        name = "partition_system_attention_indicator";
+    }
+    else if (propertyValue == "PlatformSystemAttentionIndicator")
+    {
+        name = "platform_system_attention_indicator";
+    }
+    else
+    {
+        return;
```
Comment (gtmills) [inline redfish-core/lib/oem/ibm/system_attention_indicator.hpp:86]: Log an error ?

File: redfish-core/lib/systems.hpp (hunk 1 above)
Comment (lxwinspur) [reply redfish-core/lib/systems.hpp:2901]: > Can we put these in the ifdef as well?

Ack

File: redfish-core/lib/systems.hpp [hunk 3]
```diff
@@ -3522,8 +3529,14 @@ inline void requestRoutesSystems(App& app)
...
+                        std::optional<bool> platformSAI;
+                        if (!json_util::readJson(
+                                *ibmOem, asyncResp->res, "LampTest", lampTest,
+                                "PartitionSystemAttentionIndicator",
```
[the remark this answers is not in the document]
Comment (lxwinspur) [reply redfish-core/lib/systems.hpp:2908]: > Can we put these in the ifdef as well?

Ack

File: redfish-core/lib/oem/ibm/system_attention_indicator.hpp (hunk 2 above)
Comment (lxwinspur) [reply redfish-core/lib/oem/ibm/system_attention_indicator.hpp:87]: > Log an error ?

Ack

File: redfish-core/lib/systems.hpp (hunk 1 above)
Comment (jinuthomas) [reply redfish-core/lib/systems.hpp:2901]: should the lamp test and SAI not be different and under their own #define ? In a case where we have SAI enabled and not lamp test, would we want to tread Json for lamp test ?

**Comment by lxwinspur** [comment]
> > https://github.com/ibm-openbmc/bmcweb/blob/1020/static/redfish/v1/schema/OemComputerSystem_v1.xml
> 
> But I don't see properties the properties you are implementing like PartitionSystemAttentionIndicator See a PR like https://github.com/ibm-openbmc/bmcweb/pull/130/files

I will check this

**Review by jinuthomas — APPROVED** [review]

**Comment by lxwinspur** [comment]
> 



> > https://github.com/ibm-openbmc/bmcweb/blob/1020/static/redfish/v1/schema/OemComputerSystem_v1.xml
> 
> But I don't see properties the properties you are implementing like PartitionSystemAttentionIndicator See a PR like https://github.com/ibm-openbmc/bmcweb/pull/130/files

I checked `https://github.com/ibm-openbmc/bmcweb/pull/130/files` and then run the validator and get the following results:
`WARNING - ACF not defined in schema Complex IBM Resource.OemObject (check version, spelling and casing)`

**Comment by gtmills** [comment]
jenkins run tests please

**Comment by gtmills** [comment]
Where are the schema changes?
Do you see where there are schema changes to add this new OEM property FirmwareVersion ? https://github.com/ibm-openbmc/bmcweb/pull/241/files

We need those changes for these properties

**Comment by mzipse** [comment]
@lxwinspur , can you respond to the above comments?  This is impacting one of our internal defects (SW541930).

**Comment by lxwinspur** [comment]
> Where are the schema changes? Do you see where there are schema changes to add this new OEM property FirmwareVersion ? https://github.com/ibm-openbmc/bmcweb/pull/241/files
> 
> We need those changes for these properties

Sorry for the late reply, I am working on this and will re-push today.

**Review by gtmills — APPROVED** [review]

**Pull request merged**